index/scorch/persister.go

```diff
@@ -51,6 +55,20 @@ type persisterOptions struct {
 	// PersisterNapTimeMSec amount of time to improve the chances for
 	// a healthier and heavier in-memory merging
 	PersisterNapUnderNumFiles int
+
+	// ActiveMergeMemorySegments controls the merge behavior of the
+	// persister for memory segments. If enabled, the persister will always
+	// merge the memory segments at root before flushing to disk.
+	// Once disabled, the persister will skip the merge of memory segments
+	// when there is a memory pressure detected at the application level.
+	// This helps in better adherence to the set memory quota.
+	ActiveMergeMemorySegments bool
+
+	// MemoryPressurePauseThreshold let persister to have a better leeway
+	// for prudently performing the memory merge of segments on a memory
+	// pressure situtation. The default value would be zero to favour the
+	// total skipping of memory segments merge during such a situation.
+	MemoryPressurePauseThreshold int
```

Comment:
What are meaning values for this threshold? Is it the number of in-memory segments? Or?

Reply:
It is an approximation for the application workers blocked on memory pressure.  0 meaning no app_herder blocked threads in system.

Reply:
I see, d'you think you can add a generic explanation to the commentary above, so it'd be easy to understand what one's configuring while updating the threshold.

Reply:
done!

Reply:
Some typos in the comments, but otherwise looks good to me.